source/synthDrivers/oneCore.py

```diff
@@ -344,12 +350,24 @@ def _processQueue(self):
 			log.debug("Queue empty, done processing")
 		self._isProcessing = False
 
+	def _handleSpeechFailure(self):
+		# The C++ code will log an error with details.
+		log.error("ocSpeech_speak failed!")  # so a warning beep is played when speech fails
+		try:
+			self._processQueue()
+		finally:
+			self._consecutiveSpeechFailures += 1
+			if self._consecutiveSpeechFailures >= self.MAX_CONSECUTIVE_SPEECH_FAILURES:
+				log.debugWarning("Too many consecutive speech failures, changing synth")
+				setNextSynth(self.name)

```

Comment:
I'm a bit worried about this implementation, i.e. a synth driver calling a synthDriverHandler function. Could you consider something like the braille handler does, i.e. in _writeCells, it tries to write content to a display. If that fails, it calls handleDisplayUnavailable which performs the fallback.

I think it might be best to use a new exception type for this, e.g. `ConsecutiveSPeechFailureException`. The OneCore synth can throw that exception in its speak method when it notices the self._consecutiveSpeechFailures limit has been exceeded. Then, the speech manager can catch the exception and perform the fallback.

Reply:
Handling failures here is quite tricky. I've opted to queue a synth change instead here

```
ERROR - unhandled exception (11:22:58.596) - Dummy-3 (20092):
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 232, in 'calling callback function'
  File "synthDrivers\oneCore.py", line 373, in _callback
    raise AssertionError()
AssertionError
~``